nltk/translate/ribes_score.py

```diff
@@ -0,0 +1,280 @@
+        >>> list(find_increasing_sequences(worder))
+        [(7, 8, 9, 10), (0, 1, 2, 3, 4, 5)]
+    
+    :param worder: The worder list output from word_rank_alignment
+    :param type: list(int)
+    """
+    items = iter(worder)
+    a, b = None, next(items, None)
+    result = [b]
+    while b is not None:
+        a, b = b, next(items, None)
+        if b is not None and a + 1 == b:
+            result.append(b)
+        else:
+            if len(result) > 1:
+                yield tuple(result)
+            result = [b]
+
+
+def kendall_tau(worder, normalize=True):
+    """
+    Calculates the Kendall's Tau correlation coefficient given the *worder*
+    list of word alignments from word_rank_alignment(), using the formula:
+    
+        tau = 2 * num_increasing_pairs / num_possible pairs -1
+    
+    Note that the no. of increasing pairs can be discontinuous in the *worder*
+    list and each increase sequence can be tabulated as choose(len(seq), 2) no. 
+    of increasing pair, e.g.
+    
+        >>> worder = [7, 8, 9, 10, 6, 0, 1, 2, 3, 4, 5]
+        >>> number_possible_pairs = choose(len(worder), 2)
+        >>> round(kendall_tau(worder, normalize=False),3)
+        -0.236
+        >>> round(kendall_tau(worder),3)
+        0.382
+    
+    To compare with the _ribes_kendall_tau():
+    
+        >>> worder = [0, 1, 8, 9, 10, 11, 12, 13, 14, 15, 24, 17, 18, 19, 20, 25, 26]
+        >>> round(kendall_tau(worder), 3)
+        0.265
+    
+    :param worder: The worder list output from word_rank_alignment
+    :type worder: list(int)
+    :param normalize: Flag to indicate normalization
+    :type normalize: boolean
+    """
+    worder_len = len(worder)
+    # Extract the groups of increasing/monotonic sequences.
+    increasing_sequences = find_increasing_sequences(worder)
+    # Calculate no. of increasing_pairs in *worder* list.
+    num_increasing_pairs = sum(choose(len(seq),2) for seq in increasing_sequences) 
+    # Calculate no. of possible pairs.
+    num_possible_pairs = choose(worder_len, 2)
+    # Kendall's Tau computation.
+    tau = 2 * num_increasing_pairs / num_possible_pairs -1
+    if normalize: # If normalized, the tau output falls between 0.0 to 1.0
+        return (tau + 1) /2
+    else: # Otherwise, the tau outputs falls between -1.0 to +1.0
+        return tau
+
+
+def spearman_rho(worder, normalize=True):
+    """
+    Calculates the Spearman's Rho correlation coefficient given the *worder* 
+    list of word alignment from word_rank_alignment(), using the formula:
+    
+        rho = 1 - sum(d**2) / choose(len(worder)+1, 3)  
+        
+    Given that d is the sum of difference between the *worder* list of indices
+    and the original word indices from the reference sentence.
+    
+    Using the (H0,R0) and (H5, R5) example from the paper
+    
+        >>> worder =  [7, 8, 9, 10, 6, 0, 1, 2, 3, 4, 5]
+        >>> sum((wi - i)**2 for wi, i in zip(worder, range(len(worder))))
```

Comment:
This line doesn't seem to belong in the doctest because it is about internal calculations. `worder` is also defined twice, so one of them could be removed.